core/src/consensus/tendermint/backup.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2018 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::sync::Arc;
+
+use ckey::Signature;
+use kvdb::{DBTransaction, KeyValueDB};
+use primitives::H256;
+
+use super::message::ConsensusMessage;
+use super::types::{Height, Step, View};
+use crate::db;
+
+const BACKUP_KEY: &[u8] = b"tendermint-backup";
+
+pub struct BackupView<'a> {
+    pub height: &'a Height,
+    pub view: &'a View,
+    pub step: &'a Step,
+    pub votes: &'a [ConsensusMessage],
+    pub last_seal: &'a (H256, View, Vec<Signature>),
+}
+
+pub struct BackupData {
+    pub height: Height,
+    pub view: View,
+    pub step: Step,
+    pub votes: Vec<ConsensusMessage>,
+    pub proposal: Option<H256>,
+    pub last_seal: (H256, View, Vec<Signature>),
+}
+
+pub struct Backup {
+    db: Arc<KeyValueDB>,
```

Comment:
I don't think we need to create `Backup` struct.
How about making `backup` and `load` free functions that take the reference of `KeyValueDB`?

Reply:
And I prefer `restore` to `load`.

Reply:
Good I fixed it.